### PR TITLE
[ENDPOINT] ADD-LIST-DETAIL-DELETE /activities - OT102-66

### PR DESCRIPTION
**Pull request by gabyMangoni**
OT102-66 / 123 / 124 / 125
https://alkemy-labs.atlassian.net/browse/OT102-66
-------------------------------------------------------------------------------------------------------------------------------
_**[ENDPOINT]**_

**[CREATION ] Description**
AS admin user
I WANT to register Activities
TO report them in the system

**Acceptance criteria:**
POST / activities - You must validate the existence of the fields name and content sent, to store the record in the activities table.

-------------------------------------------------------------------------------------------------------------------------------

**[LIST] Description**
HOW: admin user
I WANT: to list the Activities
TO: be able to manage the Activities

**Acceptance criteria:** 
GET a / activities. It will return the list of activities.

-------------------------------------------------------------------------------------------------------------------------------
**[DETAIL] Description**
HOW: view the details of an activity
I WANT: see the details of an activity
TO: be able to report content updates

**Acceptance criteria:** 
GET a / activities /: id. It will show all the fields of an activity based on the id sent by parameter.

-------------------------------------------------------------------------------------------------------------------------------

**[DELETE] Description**
AS admin user
I WANT to delete an existing activity
TO keep information up to date

**Acceptance criteria:** 
DELETE / activities /: id - You must validate the existence based on the id sent by parameter. If it exists, delete it

-------------------------------------------------------------------------------------------------------------------------------

**Evidence**
**Create validation:**
act-add-validacion

**Create**
act-add

**List**
act-list

**Detail**
act-detalle

**Delete**
act-delete

### CODE REVIEW

File: services/activities.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  }
+}
+
+exports.getById = async (id) => {
+  try {
+    const activityById = await Activity.findOne({
+      where: { id },
+    })
+
+    return activityById
```
Comment (JonatanGarbuyo) [inline services/activities.js:26]: acá deberías verificar si es que el activity con ese id existe.
y si no existe devolver 404.
por que sino' estarías devolviendo nada  ( o 500 server error)    y no es correcto.

**Review by dvelezroman — CHANGES_REQUESTED** [review]
No es optimo mezclar varios tickets en un solo PR.... por esta vez dejemosle asi.. la proxima.. recordar que cada ticket debe tener su branch y PR....

File: services/activities.js (hunk 1 above)
Comment (dvelezroman) [reply services/activities.js:26]: clro.. se debe verificar si existe o no...

File: services/activities.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  }
+}
+
+exports.deleteActivity = async (id) => {
+  try {
+    const activity = await Activity.findOne({
+      where: { id },
+    })
+    if (!activity) {
+      throw Error(`ID ${id} already deleted.`)
```
Comment (dvelezroman) [inline services/activities.js]: el mensaje adecuador podria ser: 
`Id ${id} no existe.`

File: routes/activities.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+const { validateRequest } = require('../middlewares')
+const isAdmin = require('../middlewares/isAdmin')
+const { activitySchema } = require('../schemas/activities')
+
+const router = new express.Router()
+
+router.get('/', get)
+router.get('/:id', getActivity)
+router.delete('/:id', isAdmin, destroy)
+router.post('/', [isAdmin, validateRequest(activitySchema)], post)
```
Comment (dvelezroman) [inline routes/activities.js:15]: excellent

**Review by JonatanGarbuyo — APPROVED** [review]

**Review by dvelezroman — APPROVED** [review]
Excellent....

**Review by joselo85 — APPROVED** [review]

**Pull request merged**